### PR TITLE
WARN introduce FutureWarning for value_counts behaviour change

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] xref #49497 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

As discussed in the dev call:
- the new behaviour would be more desirable than the current one;
- warning about it if users don't pass `name=` would be very noisy
- if we just warn for `1.5.2`, then there's at least some warning and it won't be there for too long as `2.0.0` should be out in a few months' time

Note that no warning is introduced for 
```
df.groupby(col, as_index=False).value_counts()
```
as that's the one that the other `value_counts` will conform to

### CODE REVIEW

File: doc/source/getting_started/comparison/includes/filtering.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ The above statement is simply passing a ``Series`` of ``True``/``False`` objects
 returning all rows with ``True``.
 
 .. ipython:: python
+    :okwarning:
```
Comment (jreback) [inline doc/source/getting_started/comparison/includes/filtering.rst:12]: fix the docs rather that ignoring the warnings

File: doc/source/whatsnew/v0.11.0.rst [hunk 2]
```diff
@@ -289,6 +289,7 @@ Furthermore ``datetime64[ns]`` columns are created by default, when passed datet
 (:issue:`2809`, :issue:`2810`)
 
 .. ipython:: python
+   :okwarning:
```
Comment (jreback) [inline doc/source/whatsnew/v0.11.0.rst:292]: older release notes ok to ignore (or convert to code blocks )

File: doc/source/whatsnew/v1.5.2.rst [hunk 3]
```diff
@@ -33,7 +33,7 @@ Bug fixes
 
 Other
 ~~~~~
--
+- Introduced ``FutureWarning`` notifying about behaviour change in :meth:`DataFrame.value_counts`, :meth:`Series.value_counts`, :meth:`DataFrameGroupBy.value_counts`, :meth:`SeriesGroupBy.value_counts` - the resulting series will by default now be named ``'counts'`` (or ``'proportion'`` if ``normalize=True``), and the index (if present) will be taken from the original object's name (:issue:`49497`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.2.rst:36]: this warrants a full on section and before / after on how to fix

File: pandas/core/describe.py [hunk 4]
```diff
@@ -252,7 +253,10 @@ def describe_categorical_1d(
         Ignored, but in place to unify interface.
     """
     names = ["count", "unique", "top", "freq"]
-    objcounts = data.value_counts()
+    with warnings.catch_warnings():
+        msg = "In pandas 2.0.0, the name of the resulting Series"
+        warnings.filterwarnings("ignore", msg, FutureWarning)
+        objcounts = data.value_counts()
```
Comment (jreback) [inline pandas/core/describe.py:259]: can we fix it here? the fact that u have to catch is a problem

File: pandas/core/frame.py [hunk 5]
```diff
@@ -7035,6 +7035,12 @@ def value_counts(
                     NaN            1
         dtype: int64
         """
+        warnings.warn(
+            "In pandas 2.0.0, the name of the resulting Series will be "
+            "'count' (or 'proportion' if `normalize=True`).",
+            FutureWarning,
```
Comment (jreback) [inline pandas/core/frame.py:7041]: this is going to be extremely noisy 

is there a way to do so that it doesn't show the warnjng

**Review by jreback — CHANGES_REQUESTED** [review]
this going to be extremely noisy as implemented

**Comment by MarcoGorelli** [comment]
there's conflicting reviews now 😄 https://github.com/pandas-dev/pandas/pull/49640#discussion_r1020205763

Having been through adjusting all the warnings in this PR, I am concerned about noise (`df.groupby(col).apply(lambda df: df.value_counts())` in particular would emit multiple warnings

@rhshadrach would you be OK with adding a `name` parameter to ease the deprecation cycle? The only other alternative (other than not doing this) I can think of would be to make the change without warning

File: pandas/core/frame.py (hunk 5 above)
Comment (rhshadrach) [reply pandas/core/frame.py:7041]: Responding to https://github.com/pandas-dev/pandas/pull/49640#issuecomment-1312560850 as well. 

I thought @jreback expressed the idea of having the warning (with no way to silence) on the call; it seems like I misunderstood. My preferred route would be no warning; another option is to just have a DeprecationWarning - anyone running an interactive session won't see the warning, but those testing via pytest would. While I dislike the idea of adding `name`, I'm not -1 on it.

From the linked issue on adding `name`, I'm taking some liberties here but it seems @jorisvandenbossche is +1, @jbrockmendel is -0.5, @Dr-Irv is +1, and @MarcoGorelli was -1.

Comment (MarcoGorelli) [reply pandas/core/frame.py:7041]: I presume there will be a 1.5.3 release before 2.0.0? If so, perhaps let's put this off until then, don't want to rush a decision

I wouldn't be keen on adding `name`, but I'd still prefer that to not making the change

Comment (jreback) [reply pandas/core/frame.py:7041]: if we did a DeprecationWarnjng though then this just is really a breaking change and we should just do it

so either am ok with either
- just change it in 2.0
- engineer a nice way to warn in 1.5.x but cannot be so noisy it's everywhere

**Comment by MarcoGorelli** [comment]
closing for now as it's not the right solution, will a comment to the issue with further thoughts

**Pull request closed without merging**